libqtile/widget/check_updates.py

```diff
@@ -56,29 +56,49 @@ def __init__(self, **config):
                          "Mandriva": ("urpmq --auto-select", 0)
                          }
 
-        # Check if distro name is valid.
-        try:
-            self.cmd = self.cmd_dict[self.distro][0].split()
-            self.subtr = self.cmd_dict[self.distro][1]
-        except KeyError:
-            distros = sorted(self.cmd_dict.keys())
-            logger.error(self.distro + ' is not a valid distro name. ' +
-                         'Use one of the list: ' + str(distros) + '.')
-            self.cmd = None
+        # Backwards compatibility for custom_command. Convert string to tuple.
+        if type(self.custom_command) is not tuple:  # Maybe remove this if statement after deprecation time period expires?
+            self.custom_command = (self.custom_command, 0)
+            logger.warning("string inputs to custom_command are deprecated." +
+                            "please use tuple: (\"command\",int(lines to strip from output))")
+
+        if not self.custom_command[0]:
+            # Check if distro name is valid.
+            try:
+                # logger.warning("checking updates with distro")
+                self.cmd = self.cmd_dict[self.distro][0]  # .split()
+                self.subtr = self.cmd_dict[self.distro][1]
+                # logger.warning(self.cmd)
+            except KeyError:
+                distros = sorted(self.cmd_dict.keys())
+                logger.error(self.distro + ' is not a valid distro name. ' +
+                             'Use one of the list: ' + str(distros) + '.')
+                self.cmd = None
+                self.subtr = 0
+        else:
+            # Otherwise use custom_command
+            try:
+                # logger.warning("checking updates with custom_command")
+                self.cmd = self.custom_command[0]  # .split()
+                self.subtr = self.custom_command[1]
+                # logger.warning(self.cmd)
+            except:
```

Comment:
Let's not use bare except:s here, and just let the stack trace happen. The qtile logging framework will take care of catching it and making sure it doesn't cause any problems.